### PR TITLE
:art: rename EntrepriseBaseUrl config to GitlabURL

### LINKED ISSUE #34: Setting  `Enterprise Base URL` is not appropriate for GitLab
This property is copied from `mattermost-plugin-github` and because in github all on-premise instances are entreprise grade, the name is good. But gitlab can have on-premise without etreprise lincence.

I propose to rename it `Gitlab URL` with `https://gitlab.com` as default value, without optional.

### PR DESCRIPTION
**Pull request by manland**
Fix #34

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -16,6 +16,13 @@
     "settings_schema": {
         "header": "To set up the GitLab plugin, you need to register a GitLab OAuth app here https://gitlab.com/profile/applications.",
         "settings": [
+            {
+                "key": "GitlabURL",
+                "display_name": "Gitlab Base URL",
+                "type": "text",
+                "default": "https://gitlab.com",
```
Comment (hanzei) [inline plugin.json]: :+1:

File: plugin.json [hunk 2]
```diff
@@ -16,6 +16,13 @@
     "settings_schema": {
         "header": "To set up the GitLab plugin, you need to register a GitLab OAuth app here https://gitlab.com/profile/applications.",
         "settings": [
+            {
+                "key": "GitlabURL",
+                "display_name": "Gitlab Base URL",
```
Comment (hanzei) [inline plugin.json]: Maybe `Gitlab URL` instead of `Gitlab Base URL`? 1/5

File: server/configuration.go [hunk 3]
```diff
@@ -42,6 +43,9 @@ func (c *configuration) Clone() *configuration {
 
 // IsValid checks if all needed fields are set.
 func (c *configuration) IsValid() error {
+	if _, err := url.ParseRequestURI(c.GitlabURL); err != nil {
```
Comment (hanzei) [inline server/configuration.go]: Why `url.ParseRequestURI` instead of `url.Parse`?

File: server/configuration.go [hunk 4]
```diff
@@ -42,6 +43,9 @@ func (c *configuration) Clone() *configuration {
...
+		return fmt.Errorf("Must have a valid GitLab URL")
```
Comment (hanzei) [inline server/configuration.go]: Nit: 
```suggestion
		return errors.New("Must have a valid GitLab URL")
```

File: server/configuration.go (hunk 3 above)
Comment (manland) [reply server/configuration.go]: `url.Parse` can be [relative](https://godoc.org/net/url#Parse) : `/toto/tata` is valid

File: plugin.json (hunk 2 above)
Comment (manland) [reply plugin.json]: :heavy_check_mark:

**Comment by manland** [comment]
Again... thank you very much @hanzei to take time to review all my PR :100: :1st_place_medal:

**Pull request merged**